### PR TITLE
Jwt refactor

### PR DESCRIPTION
**Pull request by BenoitSafari**
I've complety reworked the JWT verification methods.

There's now 2 hooks that handles the verifications:
- `accessVerify`
- `refreshVerify`

The access token is to be sent with protected endpoints using *bearer* method.
The refresh token is to be sent **only** for reclaiming a new access token using cookie.

Protected routes must use the `onRequest` hook like below:
```typescript
  fastify.route({
    method: "PUT",
    url: "/users",
    schema: putUserByIdSchema,
    handler: handlePutUserById,
    onRequest: [fastify.accessVerify],
  });
```

Handlers of protected routes can access "user" object using `request.user`.

I also did a small code cleanning.

### CODE REVIEW

File: api/src/modules/auth/auth.handler.ts [hunk 1]
```diff
@@ -79,57 +75,74 @@ export const handleLogin = async (request: Request, reply: Reply) => {
       throw new Error("Mot de passe incorrect.");
     }
 
+    const userObject = { 
+      id: user.id,
+      pseudo: user.pseudo,
+      mail: user.mail,
```
Comment (Julien-Goletto) [inline api/src/modules/auth/auth.handler.ts:81]: Suppress mail and avatar_url

File: api/src/modules/auth/auth.handler.ts [hunk 2]
```diff
@@ -79,57 +75,74 @@ export const handleLogin = async (request: Request, reply: Reply) => {
...
-      .setCookie("token", token, {})
-      .setCookie("refreshToken", refreshToken, {})
+      .setCookie("refresh_token", refreshToken, {
+        signed: true,
+      })
       .send({
-        user: user,
+        user: userObject,
+        token: accessToken,
         response: `Utilisateur "${pseudo}" connecté avec succés.`,
```
Comment (Julien-Goletto) [inline api/src/modules/auth/auth.handler.ts:101]: succès

File: api/src/plugins/jwt.ts [hunk 3]
```diff
@@ -20,13 +36,22 @@ const jwt: FastifyPluginCallback = async (fastify, opts, done) => {
 
   fastify.register(fastifyJwt, {
     secret: process.env.JWT_SECRET,
```
Comment (Julien-Goletto) [inline api/src/plugins/jwt.ts:38]: Utiliser deux secrets différents pour access et refresh

**Review by Julien-Goletto — APPROVED** [review]

File: api/src/modules/auth/auth.handler.ts (hunk 2 above)
Comment (BenoitSafari) [reply api/src/modules/auth/auth.handler.ts:101]: It will be corrected in a futur PR

File: api/src/modules/auth/auth.handler.ts (hunk 1 above)
Comment (BenoitSafari) [reply api/src/modules/auth/auth.handler.ts:81]: It will be corrected in a futur PR

File: api/src/plugins/jwt.ts (hunk 3 above)
Comment (BenoitSafari) [reply api/src/plugins/jwt.ts:38]: It will be corrected in a futur PR

**Pull request merged**